Fix Symfony 6.1 deprecations

willemverspyck opened the pull request:
| Q | A
| --- | ---
| Branch? | 2.x
| Bug fix? | no
| New feature? | no
| BC breaks? |no
| Deprecations? | yes
| License | MIT
| Doc | <!--new features must be documented. it is ok to first submit a draft for review and document once the general idea is accepted-->

Since Symfony 6.1 setting the $defaultName is deprecated. Prefer to use "AsCommand" attribute, but that not supported yet in PHP ^7.1, so I moved it to the "setName" method.

<!--
- Please take a moment to complete the template above by answering
- yes or no to the given questions and providing the bundle version.
-
- Afterward, replace this comment with the description of your PR.
-->

dbu reviewed (approved):
things keep changing... thanks for the fix!

dbu commented:
i merged the 2.x branch into the upcoming 3.x branch. that one has a more restrictive minimal php version requirement - if you have the time it would be great if you can do another PR for the 3.x branch to use the `AsCommand` attribute.

willemverspyck commented:
Thanks @dbu. I can add the attributes to version 3, but I see it still supports PHP 7.4 and attributes only work for PHP 8 and higher. Will support for PHP 7.4 be dropped for version 3?

dbu commented:
ah dang. given that symfony 6 only supports php 8, i think the bundle should do the same. can you also do that change in the PR to the 3.x branch?

willemverspyck commented:
I did the changes to the 3.x branche and removed support of PHP 7.4. Also removed support of Symfony 4.4, because the attribute "AsCommand" is supported since Symfony 5.3. Not sure if you want these changes for 3.x.
